contracts/token/ControlledToken.sol

```diff
@@ -29,6 +29,7 @@ contract ControlledToken is ERC20PermitUpgradeable, ControlledTokenInterface {
     virtual
     initializer
   {
+    require(address(_controller) != address(0), "ControlledToken/controller-not-zero");
```

Comment:
Use the `isContract` helper from Open Zeppelin and add some unit tests. Same remark for Ticket.
https://docs.openzeppelin.com/contracts/2.x/api/utils#Address-isContract-address-

Reply:
isContract is expensive and doesn't offer much better protection than a simple non-zero check